src/Analyser/MutatingScope.php

```diff
@@ -1957,6 +1957,187 @@ private function resolveType(Expr $node): Type
 			if ($this->reflectionProvider->hasConstant($node->name, $this)) {
 				/** @var string $resolvedConstantName */
 				$resolvedConstantName = $this->reflectionProvider->resolveConstantName($node->name, $this);
+				// core, https://www.php.net/manual/en/reserved.constants.php
+				if ($resolvedConstantName === 'PHP_VERSION') {
+					return new IntersectionType([
+						new StringType(),
+						new AccessoryNonEmptyStringType(),
+					]);
+				}
+				if ($resolvedConstantName === 'PHP_MAJOR_VERSION') {
+					return IntegerRangeType::fromInterval(5, null);
+				}
+				if ($resolvedConstantName === 'PHP_MINOR_VERSION') {
+					return IntegerRangeType::fromInterval(0, null);
```

Comment:
We might as well limit to 6 the minor version.

With one year per version, even if we go to PHP 8.7 (which is doubtful) we wouldn't have to change this before 2027. 

I'd even limit the major version to 8 and change it in a few years

Reply:
maybe `9`...

Reply:
thx for the input @dktapps, I agree and also for the sake of keeping this simple I adapted the version constants back to have open max values. Also, this could still be adapted at a later time if really needed.

Reply:
The max. values cannot be above 99, so we should go with it.

Reply:
Actually they can, for the major versions.

Reply:
PHP version 99 will not be here in less than 10 years and we can always adjust if PHP versioning will be changed :)

I really think we should go with 99 now.

Reply:
As I said before, I think this is bikeshedding. Whether it's 99 or unlimited makes no difference to anyone. I don't really care much what the end result is. But I think imposing upper limits for no reason doesn't make any sense.

https://imgs.xkcd.com/comics/workflow.png